Content.Server/Construction/AnchorableSystem.cs

```diff
@@ -92,10 +91,12 @@ private void OnAnchorComplete(EntityUid uid, AnchorableComponent component, TryA
 
             // TODO: Anchoring snaps rn anyway!
             if (component.Snap)
-                xform.Coordinates = xform.Coordinates.SnapToGrid(EntityManager, _mapManager);
+            {
+                _transform.SetCoordinates(uid, xform.Coordinates.SnapToGrid(EntityManager, _mapManager));
+            }
 
             RaiseLocalEvent(uid, new BeforeAnchoredEvent(args.User, args.Using));
-            xform.Anchored = true;
+            _transform.AnchorEntity(uid, xform);
```

Comment:
So there's more of the setters hanging around than I remembered so just add a check if the entity is already anchored as anchorentity will just exception if it is.

Reply:
I maybe misunderstood your comment, but I don't see where in `AnchorEntity` an exception is raised. Can you explain to me the problem?

Reply:
calling Add on a dictionary raises an exception if the key already exists

Reply:
Yes, I know How a dictionary works. I don't see where in the function a dictionary is used. This codebase is complex and I am a newbie. So, can you take a little bit of your time to formulate a proper answer ?

Reply:
There's over 500 prs made every month and not enough maintainers so sorry if I don't provide detailed response to every query that comes in but if you have questions the discord can provide answers.

In this case the engine doesn't check if the entity is already anchored before adding it to data structures again unlike the property setter for transformcomponent.anchored.